lib/github/deployment.js

```diff
@@ -0,0 +1,21 @@
+const { Project } = require('../models');
+const transformDeployment = require('../transforms/deployment');
+const reduceProjectKeys = require('../jira/util/reduce-project-keys');
+
+module.exports = async (context, jiraClient) => {
+  const { data: jiraPayload } = await transformDeployment(context);
+
+  if (!jiraPayload) {
+    context.log({ noop: 'no_jira_payload_deployment' }, 'Halting further execution for deployment since jiraPayload is empty');
+    return;
+  }
+
+  await jiraClient.deployment.submit(jiraPayload);
+
+  const projects = [];
+  reduceProjectKeys(jiraPayload, projects);
+
+  for (const projectKey of projects) {
+    await Project.upsert(projectKey, jiraClient.baseURL);
+  }
```

Comment:
your await is sequential which will slow down everything if you have a lot of projects.  You should look to use `Promise.all` to make it parallel or if there's a way with sequelize to bulk change projects into a single command.

Reply:
Yeah I've noticed that too but I was just following the same pattern from the implementation on the other files under this folder, such as https://github.com/integrations/jira/blob/master/lib/github/pull-request.js#L53 and https://github.com/integrations/jira/blob/master/lib/github/branch.js#L20. Shouldn't I raise a new issue/PR to fix them all instead?

Reply:
Alright, leave as is for now and create a ticket in the backlog.  Probably better to fix this in the TS version anyways :P

Reply:
Follow up [ticket](https://softwareteams.atlassian.net/browse/ARC-349) raised